packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx

```diff
@@ -201,37 +230,120 @@ export class FocusZone extends BaseComponent<IFocusZoneProps, {}> implements IFo
     return false;
   }
 
+  private _evaluateFocusBeforeRender(): void {
+    const { current: root } = this._root;
+    const doc = getDocument(root);
+
+    if (doc) {
+      const focusedElement = doc.activeElement as HTMLElement;
+
+      // Only update the index path if we are not parked on the root.
+      if (focusedElement !== root) {
+        const shouldRestoreFocus = elementContains(root, focusedElement);
+
+        this._lastIndexPath = shouldRestoreFocus ? this._getIndexPath(doc.activeElement as HTMLElement) : undefined;
+      }
+    }
+  }
+
   private _onFocus = (ev: React.FocusEvent<HTMLElement>): void => {
     const { onActiveElementChanged, doNotAllowFocusEventToPropagate, onFocusNotification } = this.props;
+    const isImmediateDescendant = this._isImmediateDescendantOfZone(ev.target as HTMLElement);
+    let newActiveElement: HTMLElement | undefined;
 
     if (onFocusNotification) {
       onFocusNotification();
     }
 
-    if (this._isImmediateDescendantOfZone(ev.target as HTMLElement)) {
-      this._activeElement = ev.target as HTMLElement;
-      this._setFocusAlignment(this._activeElement);
+    if (isImmediateDescendant) {
+      newActiveElement = ev.target as HTMLElement;
     } else {
       let parentElement = ev.target as HTMLElement;
 
       while (parentElement && parentElement !== this._root.current) {
         if (isElementTabbable(parentElement) && this._isImmediateDescendantOfZone(parentElement)) {
-          this._activeElement = parentElement;
+          newActiveElement = parentElement;
           break;
         }
         parentElement = getParent(parentElement, ALLOW_VIRTUAL_ELEMENTS) as HTMLElement;
       }
     }
 
-    if (onActiveElementChanged) {
-      onActiveElementChanged(this._activeElement as HTMLElement, ev);
+    if (newActiveElement && newActiveElement !== this._activeElement) {
+      this._activeElement = newActiveElement;
+
+      if (isImmediateDescendant) {
+        this._setFocusAlignment(this._activeElement);
+      }
+
+      if (onActiveElementChanged) {
+        onActiveElementChanged(this._activeElement as HTMLElement, ev);
+      }
     }
 
     if (doNotAllowFocusEventToPropagate) {
       ev.stopPropagation();
     }
   };
 
+  private _setParkedFocus(isParked: boolean): void {
```

Comment:
Is parked focus just when focus is at the root/highest thing?

Reply:
yes, this is a feature Thomas asked for where if you remove the last item in a zone, we put focus on the parent. Then when new focusables are added, we try and restore focus within the zone.

Reply:
will add comments.